### PR TITLE
Update ceph.in

### PR DESCRIPTION
**Pull request by cp542524698**
Update ceph.in
add help msg about --keyring or -k
Signed-off-by: pengcheng chengpeng@ztgame.com

### CODE REVIEW

**Comment by smithfarm** [comment]
This doesn't pass the "Signed-off-by" check because the commit is not signed. For more information, please read https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst

**Comment by joscollin** [comment]
@cp542524698 

`$ git commit -s` will do the trick.
`$ git commit --amend` to modify your commit.

**Comment by cp542524698** [comment]
@joscollin see it

**Comment by cp542524698** [comment]
@joscollin just a commit now;
please see it agine
[https://github.com/cp542524698/work_something/blob/master/1111.png](url)

File: src/ceph.in [hunk 1]
```diff
@@ -259,6 +259,9 @@ def parse_cmdargs(args=None, target=''):
 
     parser.add_argument('-c', '--conf', dest='cephconf',
                         help='ceph configuration file')
+
+    parser.add_argument('-k', '--keyring', dest='keyring',
```
Comment (Liuchang0812) [inline src/ceph.in]: Hi, @cp542524698 why we need this? seems we do not use `keyring` variable now.

Comment (joscollin) [inline src/ceph.in]: @Liuchang0812 I was coming to that point. I request a review from @dmick here.

Comment (cp542524698) [inline src/ceph.in]: @Liuchang0812 if one host need connect more than one ceph cluster should specify ceph.conf and ceph.keyring file

Comment (Liuchang0812) [inline src/ceph.in]: so it's a incomplete PR, you will continue work on this?

thank you joscollin

**Comment by smithfarm** [comment]
@cp542524698 Welcome to Ceph - I guess this is your first contribution? This is a friendly place and all contributors are welcome. Some of the guidelines for submitting are just recommendations and others are mandatory.

The signed-off-by line is mandatory and, as @joscollin suggested, git should be configured to add it automatically via `-s`. I think you could fix it in this case by adding `<` and `>` characters like this: `Signed-off-by: pengcheng <chengpeng@ztgame.com>`

The commit message could be improved in general, though. For example, the first line (the title, which is displayed in one-line summaries) is not descriptive enough - instead of "Update ceph.in" it could say "ceph.in: add -k option".

For example, in a comment on the PR you wrote "if one host need connect more than one ceph cluster should specify ceph.conf and ceph.keyring file" -> this is exactly the sort of thing that belongs in the commit message, because that's where you explain why the change is needed.

Maybe you could take a look at how others write their commit messages? See `git log` in master branch.

File: src/ceph.in (hunk 1 above)
Comment (tchaikov) [inline src/ceph.in]: @cp542524698 `keyring` is a ceph option, and can always be specified by `ceph.conf` using 
```ini
[client]
        keyring = /path/to/keyring
```

i don't think we need to clutter the help info with the ceph options, even they are need under variant circumstances.

Comment (cp542524698) [inline src/ceph.in]: @tchaikov @Liuchang0812 all right

**Comment by cp542524698** [comment]
@smithfarm yes, you are right;

**Comment by dmick** [comment]
Just to clarify: there are a lot of 'global' ceph options that are passed on to librados but not explicitly called out as options in the help here.  I've often wished those were better documented.  But yes, you can pass -k/--keyring and it will take effect, along with a number of other options (including anything in the config file)

**Pull request closed without merging**